### ISSUE TITLE
V1Alpha2: KFService status redesign 

### ISSUE THREAD

**Issue by yuzisun** [kubeflow/kfserving#325]
/kind feature

**Describe the solution you'd like**
v1alpha2 adds transformer and explainer endpoints, this is the proposal for redesign of the `KFService` status object.

Status Object
```go
// KFServiceStatus defines the observed state of KFService
type EndpointStatusMap map[KFServiceEndpoint]* StatusConfigurationSpec

type KFServiceStatus struct {
	duckv1beta1.Status `json:",inline"`
	URL                string                 `json:"Url,omitempty"`
	Default                   EndpointStatusMap     `json:"default,omitempty"`
	Canary                    EndpointStatusMap     `json:"canary,omitempty"`
}

// StatusConfigurationSpec describes the state of the configuration receiving traffic.
type StatusConfigurationSpec struct {
	Name     string `json:"name,omitempty"`
	Replicas int    `json:"replicas,omitempty"`
	Traffic  int    `json:"traffic,omitempty"`
}
```

Status Conditions
```go
// ConditionType represents a Service condition value
const (
	// PredictRoutesReady is set when network configuration has completed for predict endpoint verb.
	PredictRoutesReady apis.ConditionType = "PredictRoutesReady"
	// ExplainRoutesReady is set when network configuration has completed for explain endpoint verb.
	ExplainRoutesReady apis.ConditionType = "ExplainRoutesReady"
	// DefaultPredictorReady is set when default predictor has reported readiness.
	DefaultPredictorReady apis.ConditionType = "DefaultPredictorReady"
	// CanaryPredictorReady is set when canary predictor has reported readiness.
	CanaryPredictorReady apis.ConditionType = "CanaryPredictorReady"
	// DefaultExplainerReady is set when default explainer has reported readiness.
	DefaultExplainerReady apis.ConditionType = "DefaultExplainerReady"
	// CanaryExplainerReady is set when canary explainer has reported readiness.
	CanaryExplainerReady apis.ConditionType = "CanaryExplainerReady"
	// DefaultTransformerReady is set when default transformer has reported readiness.
	DefaultTransformerReady apis.ConditionType = "DefaultTransformerReady"
	// CanaryTransformerReady is set when canary transformer has reported readiness.
	CanaryTransformerReady apis.ConditionType = "CanaryTransformerReady"
)
```
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by ellis-bigelow** [comment]
This is awesome. What about transform ready, etc? Should we also have a way
to signal the entire graph is ready i.e. "ready"?

**Comment by rakelkar** [comment]
> entire graph is ready

+1 possibly EndpointReady?

**Comment by rakelkar** [comment]
what does `PredictRoutesReady` mean - is it true only when default and canary are ready? why is that useful? Also for Explainer, is it fair to assume that it can never be ready before the predictor is ready?

**Comment by rakelkar** [comment]
can close @yuzisun? implemented in #388
